### PR TITLE
Adds population dependent modular map support

### PR DESCRIPTION
**Pull request by RipGrayson**
## About The Pull Request

I've been needing to do this for a bit, essentially allows for mappers to define min and max player values needed for a modular mapping insert to be in "rotation" for a given map.
E.g I could define a LV dome to only show up with a population of between 50 and 100. 

## Why It's Good For The Game

This sort of thing has been needed for quite a while and the potential is fairly limitless. We could theoretically define places in big maps like Chigusa and Magmoor that shrink on lower pop and grow at higher pop, allowing for much more flexible maps.

## Changelog
:cl:
code: Added population dependent mapping insert support.
/:cl:

### CODE REVIEW

File: code/modules/mapping/modular_mapping.dm [hunk 1]
```diff
@@ -9,6 +9,9 @@
 	var/template_width = 0
 	///Bool for whether we want to to be spawning from the middle or to the topright of the spawner (true is centered)
 	var/keepcentered = FALSE
+	var/min_player_num //minimum player number for a modular to be added to the pool of potential modular map spawns.
```
Comment (hyper2snyper) [inline code/modules/mapping/modular_mapping.dm:12]: autodoc these two like the rest.

**Review by hyper2snyper — APPROVED** [review]
this is pretty cool

**Comment by Lumipharon** [comment]
Not directly related to the pop based mechanic, but does this functionality mean you could in theory go completely nuts and have a map with a (relative) degree of randomisation? Like shifting buildings around and that sort of thing?

**Comment by RipGrayson** [comment]
Had a talk with Mole, this isn't going to work in any real world situation. Please don't merge it while I work on a better solution.

File: code/controllers/subsystem/mapping.dm [hunk 2]
```diff
@@ -32,6 +32,8 @@ SUBSYSTEM_DEF(mapping)
 	var/groundmap_voted = FALSE
 	///If true, non-admin players will not be able to initiate a vote to change shipmap
 	var/shipmap_voted = FALSE
+	//The number of connected clients for the previous round
```
Comment (BraveMole) [inline code/controllers/subsystem/mapping.dm:35]:
```suggestion
	///The number of connected clients for the previous round
```

**Review by BraveMole — CHANGES_REQUESTED** [review]
How did you test this?

**Comment by RipGrayson** [comment]
> How did you test this?

The first iteration was tested with a modified version of Big Red that has 5 debug modular inserts that I would alter the min and max population for, along with a debug message that told me when an insert had entered the map pool and which insert it was.

The current version largely used the same system, except I could alter the .json file to report a larger number of players allowing for more extensive testing of higher population. It wasn't as extensively tested as the first version, because the core of the map selection process is the same, it's just pulling the player number from a stored .json file from the last round instead of checking for currently connected clients.

**Pull request merged**